spec/recipes/proxy_spec.rb

```diff
@@ -0,0 +1,85 @@
+# encoding: UTF-8
+#
+# Copyright 2014, Christoph Hartmann
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'spec_helper'
+
+describe 'jenkins-hardening::proxy' do
+
+  # converge
+  cached(:chef_run) do
+    ChefSpec::SoloRunner.new do |node|
+      node.automatic['root_group'] = 'root'
+    end.converge(described_recipe)
+  end
+
+  before do
+    stub_command('which nginx').and_return(true)
```

Comment:
does this work as a standalone test? the cached chef-spec run does not include the required ssl databag.